homeassistant/components/touchline/climate.py

```diff
@@ -36,39 +50,41 @@ class Touchline(ClimateDevice):
     """Representation of a Touchline device."""
 
     def __init__(self, touchline_thermostat):
-        """Initialize the climate device."""
+        """Initialize the Touchline device."""
         self.unit = touchline_thermostat
         self._name = None
         self._current_temperature = None
         self._target_temperature = None
+        self._current_operation_mode = None
+        self._preset_mode = None
 
     @property
     def supported_features(self):
         """Return the list of supported features."""
         return SUPPORT_FLAGS
 
     def update(self):
-        """Update unit attributes."""
+        """Update thermostat attributes."""
         self.unit.update()
         self._name = self.unit.get_name()
         self._current_temperature = self.unit.get_current_temperature()
         self._target_temperature = self.unit.get_target_temperature()
+        self._preset_mode = self.map_mode_touchline_hass(
+            self.unit.get_operation_mode(), self.unit.get_week_program()
+        )
 
     @property
-    def hvac_mode(self) -> str:
-        """Return hvac operation ie. heat, cool mode.
+    def hvac_mode(self):
+        """Return current HVAC mode.
 
         Need to be one of HVAC_MODE_*.
         """
         return HVAC_MODE_HEAT
 
     @property
-    def hvac_modes(self) -> List[str]:
-        """Return the list of available hvac operation modes.
-
-        Need to be a subset of HVAC_MODES.
-        """
-        return [HVAC_MODE_HEAT]
+    def hvac_modes(self):
+        """Return list of possible operation modes."""
+        return OPERATION_LIST
```

Comment:
Why is this extracted out into a another constant? It is not used in any other place?

Reply:
I think it is remnants from me trying to fix the issue with the "failed to call service climate/set_hvac_mode". I have removed the constant and readded return [HVAC_MODE_HEAT]